clouddrift/ragged.py

```diff
@@ -854,6 +854,58 @@ def unpack(
     return [unpacked[i] for i in rows]
 
 
+def index_to_row(
+    index: int | list[int] | np.ndarray,
+    rowsize: list[int] | np.ndarray | xr.DataArray,
+) -> list:
+    """Obtain a list of row indices from a list of observation indices of a ragged array.
+
+    Parameters
+    ----------
+    index : int or list or np.ndarray
+        A integer observation index or a list of observation indices of a ragged array.
+    rowsize : list or np.ndarray or xr.DataArray
+        A sequence of row sizes of a ragged array.
+
+    Returns
+    -------
+    list
+        A list of row indices.
+
+    Examples
+    --------
+    To obtain the row index of observation 5 within a ragged array of three consecutive
+    rows of sizes 2, 4, and 3:
+    >>> index_to_row(5, [2, 4, 3])
+    [1]
+
+    To obtain the row indices of observations 0, 2, and 4 within a ragged array of three consecutive
+    rows of sizes 2, 4, and 3:
+    >>> index_to_row([0, 2, 4], [2, 4, 3])
+    [0, 1, 1]
+
+    """
+    # if index is an integer, convert it to a list
+    if isinstance(index, int) | isinstance(index, float):
```

Comment:
Have we tested handling a DataArray or should we include a case here that would handle grabbing the array from the `.data` accessor?